raiden/network/proxies/token_network.py

```diff
@@ -959,20 +908,70 @@ def close(
         }
         log.debug('closeChannel called', **log_details)
 
-        checking_block = self.client.get_checking_block()
+        if signature != EMPTY_SIGNATURE:
+            canonical_identifier = CanonicalIdentifier(
+                chain_identifier=self.proxy.contract.functions.chain_id().call(),
+                token_network_address=self.address,
+                channel_identifier=channel_identifier,
+            )
+            partner_signed_data = pack_balance_proof(
+                nonce=nonce,
+                balance_hash=balance_hash,
+                additional_hash=additional_hash,
+                canonical_identifier=canonical_identifier,
+            )
+            try:
+                partner_recovered_address = recover(
```

Comment:
Added a check for the signature, this should make it easier to debug silly errors.